### PR TITLE
Align ES index-prefix parameter for jobs

### PR DESCRIPTION
**Pull request by OpenGuidou**
## Which problem is this PR solving?
Proposal to fix https://github.com/jaegertracing/jaeger/issues/4268
As it's a non-backward compatible change, let me know how you want to handle this.
I'm open to changes.

## Short description of the changes
- Align ES index-prefix flag for jobs to what's defined for other jaeger components. This allows to have a single configuration for all items.

### CODE REVIEW

File: cmd/es-index-cleaner/app/flags.go [hunk 1]
```diff
@@ -44,18 +46,23 @@ type Config struct {
...
 	flags.String(password, "", "The password required by storage")
+	flags.String(indexPrefix, "", "Index prefix")
 }
 
 // InitFromViper initializes config from viper.Viper.
 func (c *Config) InitFromViper(v *viper.Viper) {
 	c.IndexPrefix = v.GetString(indexPrefix)
+	if c.IndexPrefix == "" {
+		// try with deprecated flag
+		c.IndexPrefix = v.GetString(deprecatedIndexPrefix)
```
Comment (yurishkuro) [inline cmd/es-index-cleaner/app/flags.go:64]: this should log a warning about using deprecated flag

**Review by yurishkuro — COMMENTED** [review]
I really don't think this PR is useful. It clearly shows that the naming of the flags is all over the place and inconsistent. Fixing one single flag doesn't make the situation any better, it only introduces churn and can break people in 3m who don't notice the deprecation. I don't see any benefits in breaking people for a cosmetic change.

**Comment by OpenGuidou** [comment]
> 

We can take the opportunity to review the cli flags in those commands, there are not so many.
In my opinion it's still better to fix it at the root and risk breaking people not noticing the warning than pushing the inconsistency downwards to the rest of the ecosystem integration, like with the helm chart.

File: cmd/es-index-cleaner/app/flags.go (hunk 1 above)
Comment (OpenGuidou) [reply cmd/es-index-cleaner/app/flags.go:65]: solved

**Comment by yurishkuro** [comment]
> In my opinion it's still better to fix it at the root and risk breaking people not noticing the warning than pushing the inconsistency downwards to the rest of the ecosystem integration, like with the helm chart.

I disagree. One of the worst sins for a project to do is break users. Your better have strong reasons if you do that, and here I don't see a good reason. It would be nice if we, the maintainers, were more diligent when this inconsistency was introduced the first time, but right now is has already been pushed downwards and adopted, few years ago.

The _complexity_ is also arguable. All these scripts are narrowly scoped to ES maintenance, I don't think it's especially difficult to grasp 10 or so flags, whether they are named consistently or not.

**Comment by yurishkuro** [comment]
will be fixed differently in jaeger-v2

**Pull request closed without merging**